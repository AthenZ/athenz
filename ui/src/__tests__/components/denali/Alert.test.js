/*
 * Copyright 2020 Verizon Media
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* eslint-disable no-console */
import React from 'react';
import { fireEvent, queryByText, render } from '@testing-library/react';
import Alert from '../../../components/denali/Alert';

jest.useFakeTimers();

describe('Alert', () => {
    const onClose = jest.fn();

    beforeEach(() => {
        jest.clearAllMocks();
        jest.clearAllTimers();
    });

    it('should render nothing in closed state', () => {
        const { baseElement } = render(
            <Alert
                title='An Alert'
                description='Alert Description'
                isOpen={false}
                onClose={onClose}
            />
        );

        expect(queryByText(baseElement, 'An Alert')).toBeNull();
        expect(setTimeout).not.toHaveBeenCalled();
    });

    it('should render open state', (done) => {
        const { baseElement } = render(
            <Alert
                title='An Alert'
                description='Descriptive Alert'
                isOpen={true}
                onClose={onClose}
            />
        );

        expect(baseElement).toMatchSnapshot();
        expect(queryByText(baseElement, 'An Alert')).toBeInTheDocument();
        expect(queryByText(baseElement, 'Descriptive Alert')).not.toBeNull();
        expect(baseElement.querySelector('.animated')).toBeInTheDocument();

        // Ensure alert does not close on timer

        jest.runAllTimers();

        // Need to forward one tick so setTimeout() completely processes callback
        // before we run assertions
        setImmediate(() => {
            try {
                expect(
                    queryByText(baseElement, 'An Alert')
                ).toBeInTheDocument();
                done();
            } catch (error) {
                done.fail(error);
            }
        });
    });

    it('should render all alert types', () => {
        const { baseElement, rerender } = render(
            <Alert
                title='An Alert'
                isOpen={true}
                type='danger'
                onClose={onClose}
            />
        );

        expect(baseElement).toMatchSnapshot();
        expect(
            baseElement.querySelector('.denali-alert-danger')
        ).not.toBeNull();

        rerender(
            <Alert
                title='An Alert'
                isOpen={true}
                type='info'
                onClose={onClose}
            />
        );

        expect(baseElement).toMatchSnapshot();
        expect(baseElement.querySelector('.denali-alert-info')).not.toBeNull();

        rerender(
            <Alert
                title='An Alert'
                isOpen={true}
                type='success'
                onClose={onClose}
            />
        );

        expect(baseElement).toMatchSnapshot();
        expect(
            baseElement.querySelector('.denali-alert-success')
        ).not.toBeNull();

        rerender(
            <Alert
                title='An Alert'
                isOpen={true}
                type='warning'
                onClose={onClose}
            />
        );

        expect(baseElement).toMatchSnapshot();
        expect(
            baseElement.querySelector('.denali-alert-warning')
        ).not.toBeNull();
    });

    it('should disable animations', () => {
        const { baseElement } = render(
            <Alert
                title='An Alert'
                description='Descriptive Alert'
                isOpen={true}
                noanim
                onClose={onClose}
            />
        );

        expect(baseElement).toMatchSnapshot();
        expect(baseElement.querySelector('.animated')).not.toBeInTheDocument();
    });

    it('should render with an additional class name', () => {
        const { baseElement } = render(
            <Alert
                className='custom-class'
                title='An Alert'
                isOpen={true}
                onClose={onClose}
            />
        );
        const portal = baseElement.querySelector('.denali-alert');

        expect(portal).toHaveClass('denali-alert');
        expect(portal).toHaveClass('custom-class');
    });

    it('should handle onClose event', () => {
        const { baseElement } = render(
            <Alert title='An Alert' isOpen={true} onClose={onClose} />
        );

        fireEvent.click(baseElement.querySelector('.close-button'));

        expect(onClose).toHaveBeenCalled();
    });

    it('should handle timed close', (done) => {
        expect(setTimeout).not.toHaveBeenCalled();

        const { baseElement } = render(
            <Alert
                title='An Alert'
                isOpen={true}
                duration={5000}
                onClose={onClose}
            />
        );

        expect(queryByText(baseElement, 'An Alert')).not.toBeNull();

        expect(setTimeout).toHaveBeenCalledTimes(1);
        expect(onClose).not.toHaveBeenCalled();

        jest.runAllTimers();

        // Need to forward one tick so setTimeout() completely processes callback
        // before we run assertions
        setImmediate(() => {
            try {
                expect(onClose).toHaveBeenCalledTimes(1);
                done();
            } catch (error) {
                done.fail(error);
            }
        });
    });

    it('should set timer after Alert initially closed', (done) => {
        expect(setTimeout).not.toHaveBeenCalled();

        const { baseElement, rerender } = render(
            <Alert
                title='An Alert'
                isOpen={false}
                duration={5000}
                onClose={onClose}
            />
        );

        expect(queryByText(baseElement, 'An Alert')).toBeNull();

        expect(setTimeout).not.toHaveBeenCalled();

        rerender(
            <Alert
                title='An Alert'
                isOpen={true}
                duration={5000}
                onClose={onClose}
            />
        );

        expect(setTimeout).toHaveBeenCalledTimes(1);

        jest.runAllTimers();

        // Need to forward one tick so setTimeout() completely processes callback
        // before we run assertions
        setImmediate(() => {
            try {
                expect(onClose).toHaveBeenCalledTimes(1);
                done();
            } catch (error) {
                done.fail(error);
            }
        });
    });

    it('should handle closing Alert before timer expires', (done) => {
        expect(setTimeout).not.toHaveBeenCalled();

        const { baseElement, rerender } = render(
            <Alert
                title='An Alert'
                isOpen={true}
                duration={5000}
                onClose={onClose}
            />
        );

        expect(queryByText(baseElement, 'An Alert')).not.toBeNull();
        expect(setTimeout).toHaveBeenCalled();

        rerender(
            <Alert
                title='An Alert'
                isOpen={false}
                duration={5000}
                onClose={onClose}
            />
        );

        expect(clearTimeout).toHaveBeenCalled();

        jest.runAllTimers();

        // Need to forward one tick so setTimeout() completely processes callback
        // before we run assertions
        setImmediate(() => {
            try {
                expect(onClose).not.toHaveBeenCalled();
                done();
            } catch (error) {
                done.fail(error);
            }
        });
    });

    it('should handle unmounting with timer', (done) => {
        expect(setTimeout).not.toHaveBeenCalled();
        expect(clearTimeout).not.toHaveBeenCalled();

        const { baseElement, unmount } = render(
            <Alert
                title='An Alert'
                isOpen={true}
                duration={5000}
                onClose={onClose}
            />
        );

        expect(baseElement.querySelector('.denali-alert')).not.toBeNull();

        expect(setTimeout).toHaveBeenCalledTimes(1);
        expect(clearTimeout).not.toHaveBeenCalled();
        expect(onClose).not.toHaveBeenCalled();

        unmount();

        jest.runAllTimers();

        setImmediate(() => {
            try {
                // NOTE: For some reason, clearTimeout() actually gets called twice, but I
                // haven't been able to figure out why. Even if clearTimeout() is commented
                // out, it's still called. So just check if it was called, not how many
                // times it was called
                expect(clearTimeout).toHaveBeenCalled();
                expect(onClose).not.toHaveBeenCalled();
                done();
            } catch (error) {
                done.fail(error);
            }
        });
    });
});
