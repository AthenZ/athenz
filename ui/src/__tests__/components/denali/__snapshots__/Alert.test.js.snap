// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should disable animations 1`] = `
<body
  aria-hidden="true"
  class="ReactModal__Body--open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    class="ReactModalPortal"
  />
  <div
    class="ReactModalPortal"
  />
  <div />
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    class="ReactModalPortal"
  >
    <div
      class="ReactModal__Overlay ReactModal__Overlay--after-open css-nx0pvz-modal-overlay"
    >
      <div
        aria-describedby="alert-description"
        aria-labelledby="alert-title"
        aria-modal="true"
        class="ReactModal__Content ReactModal__Content--after-open css-pr8fc2-alert denali-alert denali-alert-info"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="left-section"
        >
          <div
            class="alert-icon"
            data-testid="alert-icon"
          >
            <svg
              class="css-1hsntjd"
              data-testid="icon"
              height="28px"
              viewBox="0 0 1024 1024"
              width="28px"
            >
              <title>
                check-circle
              </title>
              <path
                d="M512 128c212.077 0 384 171.923 384 384s-171.923 384-384 384c-212.077 0-384-171.923-384-384v0c0-212.077 171.923-384 384-384v0zM512 42.667c-259.206 0-469.333 210.128-469.333 469.333s210.128 469.333 469.333 469.333c259.206 0 469.333-210.128 469.333-469.333v0c0-259.206-210.128-469.333-469.333-469.333v0z"
              />
              <path
                d="M396.16 691.413l-126.293-126.293c-8.506-8.066-13.798-19.448-13.798-32.066 0-9.118 2.763-17.59 7.498-24.625l-0.1 0.158c7.83-10.919 20.483-17.949 34.778-17.949 11.841 0 22.556 4.824 30.286 12.613l0.003 0.003 98.133 97.707 268.373-268.373c7.549-6.884 17.635-11.1 28.706-11.1 23.564 0 42.667 19.103 42.667 42.667 0 11.017-4.176 21.060-11.031 28.63l0.032-0.036-298.667 298.667c-7.733 7.795-18.45 12.621-30.293 12.621s-22.56-4.826-30.291-12.618l-0.003-0.003z"
              />
            </svg>
          </div>
        </div>
        <div
          class="right-section"
        >
          <div
            class="title-row"
          >
            <div
              class="title"
              data-testid="alert-title"
              id="alert-title"
            >
              An Alert
            </div>
            <div
              class="close-button"
            >
              <svg
                class="css-180out3"
                data-testid="icon"
                height="28px"
                viewBox="0 0 1024 1024"
                width="28px"
              >
                <title>
                  close
                </title>
                <path
                  d="M571.733 512l226.133-224c7.901-8.071 12.777-19.131 12.777-31.33 0-9.87-3.192-18.994-8.599-26.397l0.088 0.127c-7.382-10.88-19.691-17.934-33.649-17.934-12.060 0-22.89 5.267-30.315 13.626l-0.036 0.042-226.133 226.133-226.133-226.133c-7.461-8.401-18.291-13.668-30.351-13.668-13.957 0-26.267 7.055-33.558 17.793l-0.091 0.142c-5.319 7.276-8.51 16.4-8.51 26.27 0 12.199 4.876 23.259 12.785 31.338l-0.008-0.008 226.133 224-226.133 224c-7.901 8.071-12.777 19.131-12.777 31.33 0 9.87 3.192 18.994 8.599 26.397l-0.088-0.127c7.382 10.88 19.691 17.934 33.649 17.934 12.060 0 22.89-5.267 30.315-13.626l0.036-0.042 226.133-226.133 226.133 226.133c7.461 8.401 18.291 13.668 30.351 13.668 13.957 0 26.267-7.055 33.558-17.793l0.091-0.142c5.319-7.276 8.51-16.4 8.51-26.27 0-12.199-4.876-23.259-12.785-31.338l0.008 0.008z"
                />
              </svg>
            </div>
          </div>
          <div
            class="description"
            data-testid="alert-description"
            id="alert-description"
          >
            Descriptive Alert
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Alert should render all alert types 1`] = `
<body
  aria-hidden="true"
  class="ReactModal__Body--open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    class="ReactModalPortal"
  />
  <div />
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    class="ReactModalPortal"
  >
    <div
      class="ReactModal__Overlay ReactModal__Overlay--after-open css-nx0pvz-modal-overlay"
    >
      <div
        aria-describedby="alert-description"
        aria-labelledby="alert-title"
        aria-modal="true"
        class="ReactModal__Content ReactModal__Content--after-open css-1e38fvc-alert denali-alert denali-alert-danger animated"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="left-section"
        >
          <div
            class="alert-icon"
            data-testid="alert-icon"
          >
            <svg
              class="css-1wtutiy"
              data-testid="icon"
              height="28px"
              viewBox="0 0 1024 1024"
              width="28px"
            >
              <title>
                check-circle
              </title>
              <path
                d="M512 128c212.077 0 384 171.923 384 384s-171.923 384-384 384c-212.077 0-384-171.923-384-384v0c0-212.077 171.923-384 384-384v0zM512 42.667c-259.206 0-469.333 210.128-469.333 469.333s210.128 469.333 469.333 469.333c259.206 0 469.333-210.128 469.333-469.333v0c0-259.206-210.128-469.333-469.333-469.333v0z"
              />
              <path
                d="M396.16 691.413l-126.293-126.293c-8.506-8.066-13.798-19.448-13.798-32.066 0-9.118 2.763-17.59 7.498-24.625l-0.1 0.158c7.83-10.919 20.483-17.949 34.778-17.949 11.841 0 22.556 4.824 30.286 12.613l0.003 0.003 98.133 97.707 268.373-268.373c7.549-6.884 17.635-11.1 28.706-11.1 23.564 0 42.667 19.103 42.667 42.667 0 11.017-4.176 21.060-11.031 28.63l0.032-0.036-298.667 298.667c-7.733 7.795-18.45 12.621-30.293 12.621s-22.56-4.826-30.291-12.618l-0.003-0.003z"
              />
            </svg>
          </div>
        </div>
        <div
          class="right-section"
        >
          <div
            class="title-row"
          >
            <div
              class="title"
              data-testid="alert-title"
              id="alert-title"
            >
              An Alert
            </div>
            <div
              class="close-button"
            >
              <svg
                class="css-180out3"
                data-testid="icon"
                height="28px"
                viewBox="0 0 1024 1024"
                width="28px"
              >
                <title>
                  close
                </title>
                <path
                  d="M571.733 512l226.133-224c7.901-8.071 12.777-19.131 12.777-31.33 0-9.87-3.192-18.994-8.599-26.397l0.088 0.127c-7.382-10.88-19.691-17.934-33.649-17.934-12.060 0-22.89 5.267-30.315 13.626l-0.036 0.042-226.133 226.133-226.133-226.133c-7.461-8.401-18.291-13.668-30.351-13.668-13.957 0-26.267 7.055-33.558 17.793l-0.091 0.142c-5.319 7.276-8.51 16.4-8.51 26.27 0 12.199 4.876 23.259 12.785 31.338l-0.008-0.008 226.133 224-226.133 224c-7.901 8.071-12.777 19.131-12.777 31.33 0 9.87 3.192 18.994 8.599 26.397l-0.088-0.127c7.382 10.88 19.691 17.934 33.649 17.934 12.060 0 22.89-5.267 30.315-13.626l0.036-0.042 226.133-226.133 226.133 226.133c7.461 8.401 18.291 13.668 30.351 13.668 13.957 0 26.267-7.055 33.558-17.793l0.091-0.142c5.319-7.276 8.51-16.4 8.51-26.27 0-12.199-4.876-23.259-12.785-31.338l0.008 0.008z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Alert should render all alert types 2`] = `
<body
  aria-hidden="true"
  class="ReactModal__Body--open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    class="ReactModalPortal"
  />
  <div />
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    class="ReactModalPortal"
  >
    <div
      class="ReactModal__Overlay ReactModal__Overlay--after-open css-nx0pvz-modal-overlay"
    >
      <div
        aria-describedby="alert-description"
        aria-labelledby="alert-title"
        aria-modal="true"
        class="ReactModal__Content ReactModal__Content--after-open css-pr8fc2-alert denali-alert denali-alert-info animated"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="left-section"
        >
          <div
            class="alert-icon"
            data-testid="alert-icon"
          >
            <svg
              class="css-1hsntjd"
              data-testid="icon"
              height="28px"
              viewBox="0 0 1024 1024"
              width="28px"
            >
              <title>
                check-circle
              </title>
              <path
                d="M512 128c212.077 0 384 171.923 384 384s-171.923 384-384 384c-212.077 0-384-171.923-384-384v0c0-212.077 171.923-384 384-384v0zM512 42.667c-259.206 0-469.333 210.128-469.333 469.333s210.128 469.333 469.333 469.333c259.206 0 469.333-210.128 469.333-469.333v0c0-259.206-210.128-469.333-469.333-469.333v0z"
              />
              <path
                d="M396.16 691.413l-126.293-126.293c-8.506-8.066-13.798-19.448-13.798-32.066 0-9.118 2.763-17.59 7.498-24.625l-0.1 0.158c7.83-10.919 20.483-17.949 34.778-17.949 11.841 0 22.556 4.824 30.286 12.613l0.003 0.003 98.133 97.707 268.373-268.373c7.549-6.884 17.635-11.1 28.706-11.1 23.564 0 42.667 19.103 42.667 42.667 0 11.017-4.176 21.060-11.031 28.63l0.032-0.036-298.667 298.667c-7.733 7.795-18.45 12.621-30.293 12.621s-22.56-4.826-30.291-12.618l-0.003-0.003z"
              />
            </svg>
          </div>
        </div>
        <div
          class="right-section"
        >
          <div
            class="title-row"
          >
            <div
              class="title"
              data-testid="alert-title"
              id="alert-title"
            >
              An Alert
            </div>
            <div
              class="close-button"
            >
              <svg
                class="css-180out3"
                data-testid="icon"
                height="28px"
                viewBox="0 0 1024 1024"
                width="28px"
              >
                <title>
                  close
                </title>
                <path
                  d="M571.733 512l226.133-224c7.901-8.071 12.777-19.131 12.777-31.33 0-9.87-3.192-18.994-8.599-26.397l0.088 0.127c-7.382-10.88-19.691-17.934-33.649-17.934-12.060 0-22.89 5.267-30.315 13.626l-0.036 0.042-226.133 226.133-226.133-226.133c-7.461-8.401-18.291-13.668-30.351-13.668-13.957 0-26.267 7.055-33.558 17.793l-0.091 0.142c-5.319 7.276-8.51 16.4-8.51 26.27 0 12.199 4.876 23.259 12.785 31.338l-0.008-0.008 226.133 224-226.133 224c-7.901 8.071-12.777 19.131-12.777 31.33 0 9.87 3.192 18.994 8.599 26.397l-0.088-0.127c7.382 10.88 19.691 17.934 33.649 17.934 12.060 0 22.89-5.267 30.315-13.626l0.036-0.042 226.133-226.133 226.133 226.133c7.461 8.401 18.291 13.668 30.351 13.668 13.957 0 26.267-7.055 33.558-17.793l0.091-0.142c5.319-7.276 8.51-16.4 8.51-26.27 0-12.199-4.876-23.259-12.785-31.338l0.008 0.008z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Alert should render all alert types 3`] = `
<body
  aria-hidden="true"
  class="ReactModal__Body--open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    class="ReactModalPortal"
  />
  <div />
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    class="ReactModalPortal"
  >
    <div
      class="ReactModal__Overlay ReactModal__Overlay--after-open css-nx0pvz-modal-overlay"
    >
      <div
        aria-describedby="alert-description"
        aria-labelledby="alert-title"
        aria-modal="true"
        class="ReactModal__Content ReactModal__Content--after-open css-15031y-alert denali-alert denali-alert-success animated"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="left-section"
        >
          <div
            class="alert-icon"
            data-testid="alert-icon"
          >
            <svg
              class="css-79iho6"
              data-testid="icon"
              height="28px"
              viewBox="0 0 1024 1024"
              width="28px"
            >
              <title>
                check-circle
              </title>
              <path
                d="M512 128c212.077 0 384 171.923 384 384s-171.923 384-384 384c-212.077 0-384-171.923-384-384v0c0-212.077 171.923-384 384-384v0zM512 42.667c-259.206 0-469.333 210.128-469.333 469.333s210.128 469.333 469.333 469.333c259.206 0 469.333-210.128 469.333-469.333v0c0-259.206-210.128-469.333-469.333-469.333v0z"
              />
              <path
                d="M396.16 691.413l-126.293-126.293c-8.506-8.066-13.798-19.448-13.798-32.066 0-9.118 2.763-17.59 7.498-24.625l-0.1 0.158c7.83-10.919 20.483-17.949 34.778-17.949 11.841 0 22.556 4.824 30.286 12.613l0.003 0.003 98.133 97.707 268.373-268.373c7.549-6.884 17.635-11.1 28.706-11.1 23.564 0 42.667 19.103 42.667 42.667 0 11.017-4.176 21.060-11.031 28.63l0.032-0.036-298.667 298.667c-7.733 7.795-18.45 12.621-30.293 12.621s-22.56-4.826-30.291-12.618l-0.003-0.003z"
              />
            </svg>
          </div>
        </div>
        <div
          class="right-section"
        >
          <div
            class="title-row"
          >
            <div
              class="title"
              data-testid="alert-title"
              id="alert-title"
            >
              An Alert
            </div>
            <div
              class="close-button"
            >
              <svg
                class="css-180out3"
                data-testid="icon"
                height="28px"
                viewBox="0 0 1024 1024"
                width="28px"
              >
                <title>
                  close
                </title>
                <path
                  d="M571.733 512l226.133-224c7.901-8.071 12.777-19.131 12.777-31.33 0-9.87-3.192-18.994-8.599-26.397l0.088 0.127c-7.382-10.88-19.691-17.934-33.649-17.934-12.060 0-22.89 5.267-30.315 13.626l-0.036 0.042-226.133 226.133-226.133-226.133c-7.461-8.401-18.291-13.668-30.351-13.668-13.957 0-26.267 7.055-33.558 17.793l-0.091 0.142c-5.319 7.276-8.51 16.4-8.51 26.27 0 12.199 4.876 23.259 12.785 31.338l-0.008-0.008 226.133 224-226.133 224c-7.901 8.071-12.777 19.131-12.777 31.33 0 9.87 3.192 18.994 8.599 26.397l-0.088-0.127c7.382 10.88 19.691 17.934 33.649 17.934 12.060 0 22.89-5.267 30.315-13.626l0.036-0.042 226.133-226.133 226.133 226.133c7.461 8.401 18.291 13.668 30.351 13.668 13.957 0 26.267-7.055 33.558-17.793l0.091-0.142c5.319-7.276 8.51-16.4 8.51-26.27 0-12.199-4.876-23.259-12.785-31.338l0.008 0.008z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Alert should render all alert types 4`] = `
<body
  aria-hidden="true"
  class="ReactModal__Body--open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    class="ReactModalPortal"
  />
  <div />
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    class="ReactModalPortal"
  >
    <div
      class="ReactModal__Overlay ReactModal__Overlay--after-open css-nx0pvz-modal-overlay"
    >
      <div
        aria-describedby="alert-description"
        aria-labelledby="alert-title"
        aria-modal="true"
        class="ReactModal__Content ReactModal__Content--after-open css-1q88pah-alert denali-alert denali-alert-warning animated"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="left-section"
        >
          <div
            class="alert-icon"
            data-testid="alert-icon"
          >
            <svg
              class="css-1pqq6al"
              data-testid="icon"
              height="28px"
              viewBox="0 0 1024 1024"
              width="28px"
            >
              <title>
                check-circle
              </title>
              <path
                d="M512 128c212.077 0 384 171.923 384 384s-171.923 384-384 384c-212.077 0-384-171.923-384-384v0c0-212.077 171.923-384 384-384v0zM512 42.667c-259.206 0-469.333 210.128-469.333 469.333s210.128 469.333 469.333 469.333c259.206 0 469.333-210.128 469.333-469.333v0c0-259.206-210.128-469.333-469.333-469.333v0z"
              />
              <path
                d="M396.16 691.413l-126.293-126.293c-8.506-8.066-13.798-19.448-13.798-32.066 0-9.118 2.763-17.59 7.498-24.625l-0.1 0.158c7.83-10.919 20.483-17.949 34.778-17.949 11.841 0 22.556 4.824 30.286 12.613l0.003 0.003 98.133 97.707 268.373-268.373c7.549-6.884 17.635-11.1 28.706-11.1 23.564 0 42.667 19.103 42.667 42.667 0 11.017-4.176 21.060-11.031 28.63l0.032-0.036-298.667 298.667c-7.733 7.795-18.45 12.621-30.293 12.621s-22.56-4.826-30.291-12.618l-0.003-0.003z"
              />
            </svg>
          </div>
        </div>
        <div
          class="right-section"
        >
          <div
            class="title-row"
          >
            <div
              class="title"
              data-testid="alert-title"
              id="alert-title"
            >
              An Alert
            </div>
            <div
              class="close-button"
            >
              <svg
                class="css-180out3"
                data-testid="icon"
                height="28px"
                viewBox="0 0 1024 1024"
                width="28px"
              >
                <title>
                  close
                </title>
                <path
                  d="M571.733 512l226.133-224c7.901-8.071 12.777-19.131 12.777-31.33 0-9.87-3.192-18.994-8.599-26.397l0.088 0.127c-7.382-10.88-19.691-17.934-33.649-17.934-12.060 0-22.89 5.267-30.315 13.626l-0.036 0.042-226.133 226.133-226.133-226.133c-7.461-8.401-18.291-13.668-30.351-13.668-13.957 0-26.267 7.055-33.558 17.793l-0.091 0.142c-5.319 7.276-8.51 16.4-8.51 26.27 0 12.199 4.876 23.259 12.785 31.338l-0.008-0.008 226.133 224-226.133 224c-7.901 8.071-12.777 19.131-12.777 31.33 0 9.87 3.192 18.994 8.599 26.397l-0.088-0.127c7.382 10.88 19.691 17.934 33.649 17.934 12.060 0 22.89-5.267 30.315-13.626l0.036-0.042 226.133-226.133 226.133 226.133c7.461 8.401 18.291 13.668 30.351 13.668 13.957 0 26.267-7.055 33.558-17.793l0.091-0.142c5.319-7.276 8.51-16.4 8.51-26.27 0-12.199-4.876-23.259-12.785-31.338l0.008 0.008z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Alert should render open state 1`] = `
<body
  aria-hidden="true"
  class="ReactModal__Body--open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div />
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    class="ReactModalPortal"
  >
    <div
      class="ReactModal__Overlay ReactModal__Overlay--after-open css-nx0pvz-modal-overlay"
    >
      <div
        aria-describedby="alert-description"
        aria-labelledby="alert-title"
        aria-modal="true"
        class="ReactModal__Content ReactModal__Content--after-open css-pr8fc2-alert denali-alert denali-alert-info animated"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="left-section"
        >
          <div
            class="alert-icon"
            data-testid="alert-icon"
          >
            <svg
              class="css-1hsntjd"
              data-testid="icon"
              height="28px"
              viewBox="0 0 1024 1024"
              width="28px"
            >
              <title>
                check-circle
              </title>
              <path
                d="M512 128c212.077 0 384 171.923 384 384s-171.923 384-384 384c-212.077 0-384-171.923-384-384v0c0-212.077 171.923-384 384-384v0zM512 42.667c-259.206 0-469.333 210.128-469.333 469.333s210.128 469.333 469.333 469.333c259.206 0 469.333-210.128 469.333-469.333v0c0-259.206-210.128-469.333-469.333-469.333v0z"
              />
              <path
                d="M396.16 691.413l-126.293-126.293c-8.506-8.066-13.798-19.448-13.798-32.066 0-9.118 2.763-17.59 7.498-24.625l-0.1 0.158c7.83-10.919 20.483-17.949 34.778-17.949 11.841 0 22.556 4.824 30.286 12.613l0.003 0.003 98.133 97.707 268.373-268.373c7.549-6.884 17.635-11.1 28.706-11.1 23.564 0 42.667 19.103 42.667 42.667 0 11.017-4.176 21.060-11.031 28.63l0.032-0.036-298.667 298.667c-7.733 7.795-18.45 12.621-30.293 12.621s-22.56-4.826-30.291-12.618l-0.003-0.003z"
              />
            </svg>
          </div>
        </div>
        <div
          class="right-section"
        >
          <div
            class="title-row"
          >
            <div
              class="title"
              data-testid="alert-title"
              id="alert-title"
            >
              An Alert
            </div>
            <div
              class="close-button"
            >
              <svg
                class="css-180out3"
                data-testid="icon"
                height="28px"
                viewBox="0 0 1024 1024"
                width="28px"
              >
                <title>
                  close
                </title>
                <path
                  d="M571.733 512l226.133-224c7.901-8.071 12.777-19.131 12.777-31.33 0-9.87-3.192-18.994-8.599-26.397l0.088 0.127c-7.382-10.88-19.691-17.934-33.649-17.934-12.060 0-22.89 5.267-30.315 13.626l-0.036 0.042-226.133 226.133-226.133-226.133c-7.461-8.401-18.291-13.668-30.351-13.668-13.957 0-26.267 7.055-33.558 17.793l-0.091 0.142c-5.319 7.276-8.51 16.4-8.51 26.27 0 12.199 4.876 23.259 12.785 31.338l-0.008-0.008 226.133 224-226.133 224c-7.901 8.071-12.777 19.131-12.777 31.33 0 9.87 3.192 18.994 8.599 26.397l-0.088-0.127c7.382 10.88 19.691 17.934 33.649 17.934 12.060 0 22.89-5.267 30.315-13.626l0.036-0.042 226.133-226.133 226.133 226.133c7.461 8.401 18.291 13.668 30.351 13.668 13.957 0 26.267-7.055 33.558-17.793l0.091-0.142c5.319-7.276 8.51-16.4 8.51-26.27 0-12.199-4.876-23.259-12.785-31.338l0.008 0.008z"
                />
              </svg>
            </div>
          </div>
          <div
            class="description"
            data-testid="alert-description"
            id="alert-description"
          >
            Descriptive Alert
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
